src/parser_parallel.jl

```diff
@@ -90,6 +95,45 @@ function process_and_consume_task(
     end
 end
 
+"""
+    parse_file_parallel(
+        lexer::Lexer,
+        parsing_ctx::AbstractParsingContext,
+        consume_ctx::AbstractConsumeContext,
+        chunking_ctx::ChunkingContext,
+        result_buffers::Vector{<:AbstractResultBuffer},
+        ::Type{CT}=Tuple{}
+    ) where {CT} -> Nothing
+
+    Parse the file in `lexer.io` in parallel using `chunking_ctx.nworkers` tasks. User must provide
+    a `populate_result_buffer!` method which is used to parse ingested data in `chunking_ctx.bytes`, using the
+    newline positions in `chunking_ctx.newline_positions` as row boundaries into the `result_buffers`.
+    The `consume!` method is called after each `populate_result_buffer!` call, so the user can process
+    the parsed results in parallel. No `result_buffer` is accessed by more than one task at a time.
+
+    # Arguments:
+    * `lexer`: a `NewlineLexers.Lexer` object which is used to find newline positions in the input. The
+    type of the lexer affects whether the search is quote-aware or not.
+    * `parsing_ctx`: a user-provided object which is passed to `populate_result_buffer!`
+    * `consume_ctx`: a user-provided object which is passed to `consume!`
+    * `chunking_ctx`: an internal object that is used to keep track of the current chunk of data being processed
+    * `result_buffers`: a vector of user-provided objects which are used to store the parsed results
+    * `CT`: an optional, compile-time known type which is passed to `populate_result_buffer!`
```

Comment:
can we explain what `CT` is used for? 
This `parse_file_parallel` docstring says 
> known type which is passed to `populate_result_buffer!`
and the `populate_result_buffer!` docstring says 
> object which was passed to `parse_file_parallel`
but i can't find where explains what this things is/does